File: backend/internal/models/project.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Project struct {
	ID               uuid.UUID   `json:"id"`
	DeveloperID      uuid.UUID   `json:"developer_id"`
	Title            string      `json:"title"`
	Description      string      `json:"description"`
	Completed        bool        `json:"completed"`
	FundingGoalCents int32       `json:"funding_goal_cents"`
	Milestone        string      `json:"milestone"`
	Premise          string      `json:"premise"`
	Street           string      `json:"street"`
	Locality         string      `json:"locality"`
	State            string      `json:"state"`
	Zipcode          string      `json:"zipcode"`
	Latitude         float32     `json:"latitude"`
	Longitude        float32     `json:"longitude"`
	Images           []ImageLink `json:"images"`
	CompletionDate   string      `json:"completion_date"`
}

type InvestRequestBody struct {
	Amount int32 `json:"amount"`
}

type SearchRequestBody struct {
	Query string `json:"query"`
}

type ProjectPost struct {
	ID          uuid.UUID   `json:"id"`
	CreatedAt   time.Time   `json:"created_at"`
	ProjectID   uuid.UUID   `json:"project_id"`
	Title       string      `json:"title"`
	Description string      `json:"description"`
	Images      []ImageLink `json:"images"`
}

type ImageLink struct {
	ID  uuid.UUID `json:"id"`
	Url string    `json:"url"`
}

File: frontend/src/types/project.ts
export type ImageLink = {
  id: string;
  url: string;
};

export type Project = {
  id: string;
  developerId: string;
  title: string;
  description: string;
  completed: boolean;
  funding_goal_cents: number;
  milestone: string;
  premise: string;
  street: string;
  locality: string;
  state: string;
  zipcode: string;
  latitude: number;
  longitude: number;
  images: ImageLink[];
};

export type ProjectPost = {
  id: string;
  created_at: Date;
  project_id: string;
  title: string;
  description: string;
  images: ImageLink[];
};
